src/CareTogether.Core/Engines/PolicyEvaluation/ReferralCalculations.cs

```diff
@@ -528,29 +519,38 @@ ImmutableList<DateOnly> completionDates
 
         private static IEnumerable<DateRange<Guid>> GenerateDateRanges(ImmutableList<ChildLocation> childLocations)
         {
-            DateOnly? startDate = null;
-            Guid? tag = null;
+            (DateOnly, Guid)? entry = null;
 
             foreach (var childLocation in childLocations)
             {
-                if (!startDate.HasValue && !childLocation.Paused)
+                if (!entry.HasValue && !childLocation.Paused)
+                {
+                    entry = (childLocation.Date, childLocation.ChildLocationFamilyId);
+                    continue;
+                }
+
+                if (entry.HasValue && !childLocation.Paused)
                 {
-                    startDate = childLocation.Date;
-                    tag = childLocation.ChildLocationFamilyId;
+                    yield return new DateRange<Guid>(
+                        entry.Value.Item1,
+                        childLocation.Date.AddDays(-1),
```

Comment:
Add an inline comment to explain this:
> When a child is cared for, the care duration is from the start date of childcare until the next entry's date of childcare minus one day.

What about a child changing location more than once in the same day, which can happen? This would be a negative date range.

Reply:
Step 1: Add a unit test for the multiple-changes-in-one-day scenario (start w/ parent, go to volunteer, go to another volunteer, go back to the first volunteer, then go to the parent). That will introduce some fun wrinkles. 😅